teacher.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(id, specialization, name = 'Unknown', age = 0)
+    super(id, name, age)
+    @specialization = specialization
+  end
```

Comment:
- [ ] As I mentioned in the previous comment that only `parent_permission` and `name` can have default values, kindly ensure that you also update the above `initialize` method in order to correctly match the project requirements. After you have used the `Random` method to generate `id`s you don't need to list it again as `parameter`. I should look similar to this one below:
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission: true)  
    super(age, name, parent_permission)
    @specialization = specialization
  end
```